### PR TITLE
NFT on other chains

### PR DESCRIPTION
**Pull request by apaillier-ledger**
## Description

Part of the EVM framework changes.

* Fixes wording for NFT allowance on Stax
* NFT Ragger tests

## Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [x] Tests
- [ ] Documentation
- [ ] Other (for changes that might not fit in any category)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -147,13 +147,24 @@ ifneq ($(BYPASS_SIGNATURES),0)
 DEFINES += HAVE_BYPASS_SIGNATURES
 endif
 
+# Enable the SET_PLUGIN test key
+SET_PLUGIN_TEST_KEY:=0
```
Comment (fbeutin-ledger) [inline Makefile:151]: Real Makefile conditional assignment is `?=`
Let's use it everywhere from now on

File: Makefile [hunk 2]
```diff
@@ -147,13 +147,24 @@ ifneq ($(BYPASS_SIGNATURES),0)
...
+ifneq ($(SET_PLUGIN_TEST_KEY),0)
+DEFINES += HAVE_SET_PLUGIN_TEST_KEY
```
Comment (fbeutin-ledger) [inline Makefile:153]: Add a warning log

File: Makefile [hunk 3]
```diff
@@ -147,13 +147,24 @@ ifneq ($(BYPASS_SIGNATURES),0)
...
+# Enable the NFT test key
+NFT_TEST_KEY:=0
+ifneq ($(NFT_TEST_KEY),0)
+DEFINES += HAVE_NFT_TEST_KEY
+endif
+# Enable the NFT staging key
+NFT_STAGING_KEY:=0
+ifneq ($(NFT_STAGING_KEY),0)
+DEFINES += HAVE_NFT_STAGING_KEY
 endif
```
Comment (fbeutin-ledger) [inline Makefile:168]: Same 2 remarks x2

File: Makefile [hunk 4]
```diff
@@ -169,12 +180,12 @@ endif
...
-CAL_CI_KEY:=0
+CAL_STAGING_KEY:=0
 ifneq ($(CAL_TEST_KEY),0)
 DEFINES += HAVE_CAL_TEST_KEY
 endif
-ifneq ($(CAL_CI_KEY),0)
-DEFINES += HAVE_CAL_CI_KEY
+ifneq ($(CAL_STAGING_KEY),0)
+DEFINES += HAVE_CAL_STAGING_KEY
 endif
```
Comment (fbeutin-ledger) [inline Makefile:189]: Same 2 remarks x2

File: Makefile [hunk 5]
```diff
@@ -169,12 +180,12 @@ endif
 
 # CryptoAssetsList key
 CAL_TEST_KEY:=0
-CAL_CI_KEY:=0
+CAL_STAGING_KEY:=0
```
Comment (fbeutin-ledger) [inline Makefile:183]: Error if both are not 0

File: src_features/provideNFTInformation/cmd_provideNFTInfo.c [hunk 6]
```diff
@@ -19,7 +19,7 @@
 #define MIN_DER_SIG_SIZE      67
 #define MAX_DER_SIG_SIZE      72
 
-#define TEST_NFT_METADATA_KEY 0
+#define STAG_NFT_METADATA_KEY 0
```
Comment (fbeutin-ledger) [inline src_features/provideNFTInformation/cmd_provideNFTInfo.c:22]: STAGE ?

File: Makefile [hunk 7]
```diff
@@ -147,13 +147,24 @@ ifneq ($(BYPASS_SIGNATURES),0)
...
-NFT_TESTING_KEY:=0
-ifneq ($(NFT_TESTING_KEY),0)
-DEFINES += HAVE_NFT_TESTING_KEY
+# Enable the NFT test key
+NFT_TEST_KEY:=0
+ifneq ($(NFT_TEST_KEY),0)
+DEFINES += HAVE_NFT_TEST_KEY
+endif
+# Enable the NFT staging key
+NFT_STAGING_KEY:=0
```
Comment (fbeutin-ledger) [inline Makefile:165]: Erorr if both are true

File: src_features/provideNFTInformation/cmd_provideNFTInfo.c [hunk 8]
```diff
@@ -144,9 +150,11 @@ void handleProvideNFTInformation(uint8_t p1,
     // this prints raw data, so to have a more meaningful print, display
     // the buffer before the endianness swap
     PRINTF("ChainID: %.*H\n", sizeof(chainId), (workBuffer + offset));
-    if ((chainConfig->chainId != 0) && (chainConfig->chainId != chainId)) {
-        PRINTF("Chain ID token mismatch\n");
-        THROW(0x6A80);
+    if (!((chainConfig->chainId == ETHEREUM_MAINNET_CHAINID) &&
+          chain_is_ethereum_compatible(&chainId)) &&
+        (chainId != chainConfig->chainId)) {
```
Comment (fbeutin-ledger) [inline src_features/provideNFTInformation/cmd_provideNFTInfo.c:155]: I've read several time and still don't understand the condition, maybe improve or add the clear text equivalent above

File: Makefile (hunk 7 above)
Comment (fbeutin-ledger) [inline Makefile:165]: Comments on what this key is and when to use it would be nice

File: src_features/setPlugin/cmd_setPlugin.c [hunk 9]
```diff
@@ -173,9 +179,11 @@ void handleSetPlugin(uint8_t p1,
     // this prints raw data, so to have a more meaningful print, display
     // the buffer before the endianness swap
     PRINTF("ChainID: %.*H\n", sizeof(chainId), (workBuffer + offset));
-    if ((chainConfig->chainId != 0) && (chainConfig->chainId != chainId)) {
-        PRINTF("Chain ID token mismatch\n");
-        THROW(0x6A80);
+    if (!((chainConfig->chainId == ETHEREUM_MAINNET_CHAINID) &&
```
Comment (fbeutin-ledger) [inline src_features/setPlugin/cmd_setPlugin.c:182]: Same, unclear

File: src_plugins/erc1155/erc1155_plugin.c [hunk 10]
```diff
@@ -108,7 +108,12 @@ static void handle_query_contract_id(void *parameters) {
 
     switch (context->selectorIndex) {
         case SET_APPROVAL_FOR_ALL:
+#ifdef HAVE_NBGL
+            strlcpy(msg->version, "manage", msg->versionLength);
+            strlcat(msg->name, " allowance", msg->versionLength);
```
Comment (fbeutin-ledger) [inline src_plugins/erc1155/erc1155_plugin.c:113]: `msg->versionLength` is probably wrong ?

File: src_plugins/erc721/erc721_plugin.c [hunk 11]
```diff
@@ -116,7 +116,12 @@ static void handle_query_contract_id(void *parameters) {
     switch (context->selectorIndex) {
         case SET_APPROVAL_FOR_ALL:
         case APPROVE:
+#ifdef HAVE_NBGL
+            strlcpy(msg->version, "manage", msg->versionLength);
+            strlcat(msg->name, " allowance", msg->versionLength);
```
Comment (fbeutin-ledger) [inline src_plugins/erc721/erc721_plugin.c:121]: Same, either something like `msg->nameLength` or `msg->versionLength - xyz` ?

File: src_features/provideNFTInformation/cmd_provideNFTInfo.c (hunk 8 above)
Comment (apaillier-ledger) [reply src_features/provideNFTInformation/cmd_provideNFTInfo.c]: It is now much simpler. :+1:

**Review by fbeutin-ledger — DISMISSED** [review]

**Review by fbeutin-ledger — DISMISSED** [review]

**Pull request merged**